pkg/skuba/actions/addon/upgrade/apply.go

```diff
@@ -36,6 +36,29 @@ func Apply(client clientset.Interface) error {
 	if err != nil {
 		return err
 	}
+
+	clusterConfiguration, err := kubeadm.GetClusterConfiguration(client)
```

Comment:
The same validation code used in `apply` and `plan`. I would suggest to have a common validation for this to reuse.

Reply:
Well, the validation already has the function `CheckLocalAddonsBaseManifests` to achieve it.

The only way to have a more common code is the wrap the error message to a single function/variable but IMO it's minor. Or do you have any better ideas :bow: ? 

```
        if !match {
		fmt.Println("Current local addons cluster folder configuration is out-of-date.")
		fmt.Println("Please run \"skuba addon refresh localconfig\" before you perform addon upgrade.")
		return nil
	}
```

Reply:
Yes, it's minor, and just checked the code again where some variables are used before/after the context, so the way right now should be fine even there is some little duplication. It's fine for me now ;)